bindings/python/iota_sdk/types/block_issuer_key.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 IOTA Stiftung
+# SPDX-License-Identifier: Apache-2.0
+
+from enum import IntEnum
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, TypeAlias
+from iota_sdk.types.common import HexStr, json
+
+
+class BlockIssuerKeyType(IntEnum):
+    """BlockIssuerKey type variants.
+
+     Attributes:
+        ED25519 (0): Ed25519 block issuer key.
+    """
+    ED25519 = 0
+
+
+@json
+@dataclass
+class Ed25519BlockIssuerKey:
+    """An Ed25519 block issuer key.
```

Comment:
```suggestion
    """A Block Issuer Key derived from an Ed25519 Public Key.
```

Reply:
Why "derived"? It's a public key

Reply:
its what the docs say 🏃‍♂️ in `type` of "block issuers key anyof": `A Block Issuer Key derived from an Ed25519 Public Key`.
Lets suggest a change then

<html>
<body>
<!--StartFragment--><html lang="en" data-color-mode="dark" data-light-theme="light" data-dark-theme="dark" data-a11y-animated-images="system" data-a11y-link-underlines="false" data-turbo-loaded="" style="box-sizing: border-box; font-size: 16px; font-family: sans-serif; text-size-adjust: 100%; --h00-size-mobile: 2.5rem; --h0-size-mobile: 2rem; --h1-size-mobile: 1.625rem; --h2-size-mobile: 1.375rem; --h3-size-mobile: 1.125rem; --h00-size: 3rem; --h0-size: 2.5rem; --h1-size: 2rem; --h2-size: 1.5rem; --h3-size: 1.25rem; --h4-size: 1rem; --h5-size: 0.875rem; --h6-size: 0.75rem; --body-font-size: 0.875rem; --font-size-small: 0.75rem; --base-size-4: 0.25rem; --base-size-8: 0.5rem; --base-size-12: 0.75rem; --base-size-16: 1rem; --base-size-20: 1.25rem; --base-size-24: 1.5rem; --base-size-28: 1.75rem; --base-size-32: 2rem; --base-size-36: 2.25rem; --base-size-40: 2.5rem; --base-size-44: 2.75rem; --base-size-48: 3rem; --base-size-64: 4rem; --base-size-80: 5rem; --base-size-96: 6rem; --base-size-112: 7rem; --base-size-128: 8rem; --base-text-weight-light: 300; --base-text-weight-normal: 400; --base-text-weight-medium: 500; --base-text-weight-semibold: 600; --boxShadow-thin: inset 0 0 0 max(1px, 0.0625rem); --boxShadow-thick: inset 0 0 0 max(2px, 0.125rem); --boxShadow-thicker: inset 0 0 0 max(4px, 0.25rem); --borderWidth-thin: max(1px, 0.0625rem); --borderWidth-thick: max(2px, 0.125rem); --borderWidth-thicker: max(4px, 0.25rem); --borderRadius-small: 0.1875rem; --borderRadius-medium: 0.375rem; --borderRadius-large: 0.75rem; --borderRadius-full: 624.9375rem; --outline-focus-offset: -0.125rem; --outline-focus-width: 0.125rem; --breakpoint-xsmall: 20rem; --breakpoint-small: 34rem; --breakpoint-medium: 48rem; --breakpoint-large: 63.25rem; --breakpoint-xlarge: 80rem; --breakpoint-xxlarge: 90rem; --control-minTarget-auto: 1rem; --controlStack-small-gap-auto: 0.5rem; --controlStack-medium-gap-auto: 0.5rem; --control-minTarget-fine: 1rem; --control-minTarget-coarse: 2.75rem; --control-xsmall-size: 1.5rem; --control-xsmall-lineBoxHeight: 1.25rem; --control-xsmall-paddingBlock: 0.125rem; --control-xsmall-paddingInline-condensed: 0.25rem; --control-xsmall-paddingInline-normal: 0.5rem; --control-xsmall-paddingInline-spacious: 0.75rem; --control-xsmall-gap: 0.25rem; --control-small-size: 1.75rem; --control-small-lineBoxHeight: 1.25rem; --control-small-paddingBlock: 0.25rem; --control-small-paddingInline-condensed: 0.5rem; --control-small-paddingInline-normal: 0.75rem; --control-small-gap: 0.25rem; --control-medium-size: 2rem; --control-medium-lineBoxHeight: 1.25rem; --control-medium-paddingBlock: 0.375rem; --control-medium-paddingInline-condensed: 0.5rem; --control-medium-paddingInline-normal: 0.75rem; --control-medium-paddingInline-spacious: 1rem; --control-medium-gap: 0.5rem; --control-large-size: 2.5rem; --control-large-lineBoxHeight: 1.25rem; --control-large-paddingBlock: 0.625rem; --control-large-paddingInline-normal: 0.75rem; --control-large-paddingInline-spacious: 1rem; --control-large-gap: 0.5rem; --control-xlarge-size: 3rem; --control-xlarge-lineBoxHeight: 1.25rem; --control-xlarge-paddingBlock: 0.875rem; --control-xlarge-paddingInline-normal: 0.75rem; --control-xlarge-paddingInline-spacious: 1rem; --control-xlarge-gap: 0.5rem; --controlStack-small-gap-condensed: 0.5rem; --controlStack-small-gap-spacious: 1rem; --controlStack-medium-gap-condensed: 0.5rem; --controlStack-medium-gap-spacious: 0.75rem; --controlStack-large-gap-auto: 0.5rem; --controlStack-large-gap-condensed: 0.5rem; --controlStack-large-gap-spacious: 0.75rem; --space-xxsmall: 0.125rem; --space-xsmall: 0.25rem; --space-small: 0.375rem; --space-medium: 0.5rem; --space-large: 0.75rem; --space-xlarge: 1rem; --stack-padding-condensed: 0.5rem; --stack-padding-normal: 1rem; --stack-padding-spacious: 1.5rem; --stack-gap-condensed: 0.5rem; --stack-gap-normal: 1rem; --stack-gap-spacious: 1.5rem; --overlay-width-xsmall: 12rem; --overlay-width-small: 20rem; --overlay-width-medium: 30rem; --overlay-width-large: 40rem; --overlay-width-xlarge: 60rem; --overlay-height-small: 16rem; --overlay-height-medium: 20rem; --overlay-height-large: 27rem; --overlay-height-xlarge: 37.5rem; --overlay-padding-normal: 1rem; --overlay-padding-condensed: 0.5rem; --overlay-paddingBlock-condensed: 0.25rem; --overlay-paddingBlock-normal: 0.75rem; --overlay-borderRadius: 0.375rem; --text-codeInline-size: 0.9285em; --text-codeBlock-lineHeight: 1.5385; --text-codeBlock-size: 0.8125rem; --text-caption-lineHeight: 1.3333; --text-caption-size: 0.75rem; --text-body-lineHeight-small: 1.6666; --text-body-lineHeight-medium: 1.4285; --text-body-lineHeight-large: 1.5; --text-body-size-small: 0.75rem; --text-body-size-medium: 0.875rem; --text-body-size-large: 1rem; --text-subtitle-lineHeight: 1.6; --text-subtitle-size: 1.25rem; --text-title-lineHeight-small: 1.5; --text-title-lineHeight-medium: 1.6; --text-title-lineHeight-large: 1.5; --text-title-size-small: 1rem; --text-title-size-medium: 1.25rem; --text-title-size-large: 2rem; --text-display-lineHeight: 1.4; --text-display-size: 2.5rem; --text-display-lineBoxHeight: 3.5rem; --fontStack-monospace: ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace; --fontStack-sansSerif: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'; --fontStack-system: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Noto Sans', Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji'; --text-codeInline-weight: 400; --text-codeBlock-weight: 400; --text-caption-weight: 400; --text-body-weight: 400; --text-subtitle-weight: 400; --text-title-weight-small: 600; --text-title-weight-medium: 600; --text-title-weight-large: 600; --text-display-weight: 500; --text-codeInline-shorthand: var(--text-codeInline-weight) var(--text-codeInline-size) var(--fontStack-monospace); --text-codeBlock-shorthand: var(--text-codeBlock-weight) var(--text-codeBlock-size)/var(--text-codeBlock-lineHeight) var(--fontStack-monospace); --text-caption-shorthand: var(--text-caption-weight) var(--text-caption-size)/var(--text-caption-lineHeight) var(--fontStack-sansSerif); --text-body-shorthand-small: var(--text-body-weight) var(--text-body-size-small)/var(--text-body-lineHeight-small) var(--fontStack-sansSerif); --text-body-shorthand-medium: var(--text-body-weight) var(--text-body-size-medium)/var(--text-body-lineHeight-medium) var(--fontStack-sansSerif); --text-body-shorthand-large: var(--text-body-weight) var(--text-body-size-large)/var(--text-body-lineHeight-large) var(--fontStack-sansSerif); --text-subtitle-shorthand: var(--text-subtitle-weight) var(--text-subtitle-size)/var(--text-subtitle-lineHeight) var(--fontStack-sansSerif); --text-title-shorthand-small: var(--text-title-weight-small) var(--text-title-size-small)/var(--text-title-lineHeight-small) var(--fontStack-sansSerif); --text-title-shorthand-medium: var(--text-title-weight-medium) var(--text-title-size-medium)/var(--text-title-lineHeight-medium) var(--fontStack-sansSerif); --text-title-shorthand-large: var(--text-title-weight-large) var(--text-title-size-large)/var(--text-title-lineHeight-large) var(--fontStack-sansSerif); --text-display-shorthand: var(--text-display-weight) var(--text-display-size)/var(--text-display-lineHeight) var(--fontStack-sansSerif); color-scheme: dark; color: var(--fgColor-default, var(--color-fg-default)); background-color: var(--bgColor-default, var(--color-canvas-default)); --Layout-pane-width: 220px; --Layout-content-width: 100%; --Layout-template-columns: 1fr var(--Layout-pane-width); --Layout-template-areas: &quot;content pane&quot;; --Layout-column-gap: 16px; --Layout-row-gap: 16px; --Layout-outer-spacing-x: 0px; --Layout-outer-spacing-y: 0px; --Layout-inner-spacing-min: 0px; --Layout-inner-spacing-max: 0px; --actionListContent-paddingBlock: var(--control-medium-paddingBlock,0.375rem); --duration-fast: 80ms; --easing-easeInOut: cubic-bezier(0.65,0,0.35,1); --color-social-reaction-bg-hover: var(--color-scale-gray-7); --color-social-reaction-bg-reacted-hover: var(--color-scale-blue-8); --color-workflow-card-connector: var(--color-scale-gray-5); --color-workflow-card-connector-bg: var(--color-scale-gray-5); --color-workflow-card-connector-inactive: var(--borderColor-default, var(--color-border-default)); --color-workflow-card-connector-inactive-bg: var(--borderColor-default, var(--color-border-default)); --color-workflow-card-connector-highlight: var(--color-scale-blue-5); --color-workflow-card-connector-highlight-bg: var(--color-scale-blue-5); --color-workflow-card-bg: var(--color-scale-gray-7); --color-workflow-card-inactive-bg: var(--bgColor-inset, var(--color-canvas-inset)); --color-workflow-card-header-shadow: rgba(27, 31, 35, 0.04); --color-workflow-card-progress-complete-bg: var(--color-scale-blue-5); --color-workflow-card-progress-incomplete-bg: var(--color-scale-gray-6); --color-discussions-state-answered-icon: var(--color-scale-green-3); --color-bg-discussions-row-emoji-box: var(--color-scale-gray-6); --color-notifications-button-text: var(--color-scale-white); --color-notifications-button-hover-text: var(--color-scale-white); --color-notifications-button-hover-bg: var(--color-scale-blue-4); --color-notifications-row-read-bg: var(--color-canvas-default); --color-notifications-row-bg: var(--bgColor-muted, var(--color-canvas-subtle)); --color-icon-directory: var(--fgColor-muted, var(--color-fg-muted)); --color-checks-step-error-icon: var(--color-scale-red-4); --color-calendar-halloween-graph-day-L1-bg: #631c03; --color-calendar-halloween-graph-day-L2-bg: #bd561d; --color-calendar-halloween-graph-day-L3-bg: #fa7a18; --color-calendar-halloween-graph-day-L4-bg: #fddf68; --color-calendar-winter-graph-day-L1-bg: #0A3069; --color-calendar-winter-graph-day-L2-bg: #0969DA; --color-calendar-winter-graph-day-L3-bg: #54AEFF; --color-calendar-winter-graph-day-L4-bg: #B6E3FF; --color-calendar-graph-day-bg: var(--color-scale-gray-8); --color-calendar-graph-day-border: rgba(27, 31, 35, 0.06); --color-calendar-graph-day-L1-bg: #0e4429; --color-calendar-graph-day-L2-bg: #006d32; --color-calendar-graph-day-L3-bg: #26a641; --color-calendar-graph-day-L4-bg: #39d353; --color-calendar-graph-day-L1-border: rgba(255, 255, 255, 0.05); --color-calendar-graph-day-L2-border: rgba(255, 255, 255, 0.05); --color-calendar-graph-day-L3-border: rgba(255, 255, 255, 0.05); --color-calendar-graph-day-L4-border: rgba(255, 255, 255, 0.05); --color-user-mention-fg: var(--color-scale-yellow-0); --color-user-mention-bg: var(--color-scale-yellow-8); --color-dashboard-feed-bg: var(--color-scale-gray-9); --color-mktg-btn-shadow-outline: rgba(255, 255, 255, 0.25) 0 0 0 1px inset; --color-marketing-icon-primary: var(--color-scale-blue-2); --color-marketing-icon-secondary: var(--color-scale-blue-5); --color-project-header-bg: var(--color-scale-gray-9); --color-project-sidebar-bg: var(--color-scale-gray-8); --color-project-gradient-in: var(--color-scale-gray-8); --color-project-gradient-out: rgba(22, 27, 34, 0); --color-diff-blob-selected-line-highlight-mix-blend-mode: screen; --color-checks-donut-error: var(--color-scale-red-4); --color-checks-donut-pending: var(--color-scale-yellow-3); --color-checks-donut-success: var(--color-scale-green-4); --color-checks-donut-neutral: var(--color-scale-gray-3); --color-text-white: var(--color-scale-white); --line-number-cell-width: 40px; --diff-line-minimum-height: 24px; --border-color-iteration-1: #363940; --border-color-iteration-2: #3F434B; --border-color-iteration-3: #4B5159; --control-borderColor-rest: var(--border-color-iteration-4, var(--border-color-iteration-3, var(--border-color-iteration-2, var(--border-color-iteration-1)))) !important; --color-canvas-default-transparent: rgba(13,17,23,0); --color-page-header-bg: #0d1117; --color-diff-blob-addition-num-text: #e6edf3; --color-diff-blob-addition-fg: #e6edf3; --color-diff-blob-addition-num-bg: rgba(63,185,80,0.3); --color-diff-blob-addition-line-bg: rgba(46,160,67,0.15); --color-diff-blob-addition-word-bg: rgba(46,160,67,0.4); --color-diff-blob-deletion-num-text: #e6edf3; --color-diff-blob-deletion-fg: #e6edf3; --color-diff-blob-deletion-num-bg: rgba(248,81,73,0.3); --color-diff-blob-deletion-line-bg: rgba(248,81,73,0.1); --color-diff-blob-deletion-word-bg: rgba(248,81,73,0.4); --color-diff-blob-hunk-num-bg: rgba(56,139,253,0.4); --color-diff-blob-expander-icon: #7d8590; --color-diffstat-deletion-border: rgba(240,246,252,0.1); --color-diffstat-addition-border: rgba(240,246,252,0.1); --color-diffstat-addition-bg: #3fb950; --color-search-keyword-hl: rgba(210,153,34,0.4); --color-prettylights-syntax-comment: #8b949e; --color-prettylights-syntax-constant: #79c0ff; --color-prettylights-syntax-entity: #d2a8ff; --color-prettylights-syntax-storage-modifier-import: #c9d1d9; --color-prettylights-syntax-entity-tag: #7ee787; --color-prettylights-syntax-keyword: #ff7b72; --color-prettylights-syntax-string: #a5d6ff; --color-prettylights-syntax-variable: #ffa657; --color-prettylights-syntax-brackethighlighter-unmatched: #f85149; --color-prettylights-syntax-invalid-illegal-text: #f0f6fc; --color-prettylights-syntax-invalid-illegal-bg: #8e1519; --color-prettylights-syntax-carriage-return-text: #f0f6fc; --color-prettylights-syntax-carriage-return-bg: #b62324; --color-prettylights-syntax-string-regexp: #7ee787; --color-prettylights-syntax-markup-list: #f2cc60; --color-prettylights-syntax-markup-heading: #1f6feb; --color-prettylights-syntax-markup-italic: #c9d1d9; --color-prettylights-syntax-markup-bold: #c9d1d9; --color-prettylights-syntax-markup-deleted-text: #ffdcd7; --color-prettylights-syntax-markup-deleted-bg: #67060c; --color-prettylights-syntax-markup-inserted-text: #aff5b4; --color-prettylights-syntax-markup-inserted-bg: #033a16; --color-prettylights-syntax-markup-changed-text: #ffdfb6; --color-prettylights-syntax-markup-changed-bg: #5a1e02; --color-prettylights-syntax-markup-ignored-text: #c9d1d9; --color-prettylights-syntax-markup-ignored-bg: #1158c7; --color-prettylights-syntax-meta-diff-range: #d2a8ff; --color-prettylights-syntax-brackethighlighter-angle: #8b949e; --color-prettylights-syntax-sublimelinter-gutter-mark: #484f58; --color-prettylights-syntax-constant-other-reference-link: #a5d6ff; --color-codemirror-text: #e6edf3; --color-codemirror-bg: #0d1117; --color-codemirror-gutters-bg: #0d1117; --color-codemirror-guttermarker-text: #0d1117; --color-codemirror-guttermarker-subtle-text: #6e7681; --color-codemirror-linenumber-text: #7d8590; --color-codemirror-cursor: #e6edf3; --color-codemirror-selection-bg: rgba(56,139,253,0.4); --color-codemirror-activeline-bg: rgba(110,118,129,0.1); --color-codemirror-matchingbracket-text: #e6edf3; --color-codemirror-lines-bg: #0d1117; --color-codemirror-syntax-comment: #8b949e; --color-codemirror-syntax-constant: #79c0ff; --color-codemirror-syntax-entity: #d2a8ff; --color-codemirror-syntax-keyword: #ff7b72; --color-codemirror-syntax-storage: #ff7b72; --color-codemirror-syntax-string: #a5d6ff; --color-codemirror-syntax-support: #79c0ff; --color-codemirror-syntax-variable: #ffa657; --color-checks-bg: #010409; --color-checks-run-border-width: 1px; --color-checks-container-border-width: 1px; --color-checks-text-primary: #e6edf3; --color-checks-text-secondary: #7d8590; --color-checks-text-link: #2f81f7; --color-checks-btn-icon: #7d8590; --color-checks-btn-hover-icon: #e6edf3; --color-checks-btn-hover-bg: rgba(110,118,129,0.1); --color-checks-input-text: #7d8590; --color-checks-input-placeholder-text: #6e7681; --color-checks-input-focus-text: #e6edf3; --color-checks-input-bg: #161b22; --color-checks-input-shadow: 0 0 0 1px (obj) => (0, get_1.default)(obj, path); --color-checks-dropdown-text: #e6edf3; --color-checks-dropdown-bg: #161b22; --color-checks-dropdown-border: #30363d; --color-checks-dropdown-shadow: rgba(1,4,9,0.3); --color-checks-dropdown-hover-text: #e6edf3; --color-checks-dropdown-hover-bg: rgba(110,118,129,0.1); --color-checks-dropdown-btn-hover-text: #e6edf3; --color-checks-dropdown-btn-hover-bg: rgba(110,118,129,0.1); --color-checks-scrollbar-thumb-bg: rgba(110,118,129,0.4); --color-checks-header-label-text: #7d8590; --color-checks-header-label-open-text: #e6edf3; --color-checks-header-border: #21262d; --color-checks-header-icon: #7d8590; --color-checks-line-text: #7d8590; --color-checks-line-num-text: #6e7681; --color-checks-line-timestamp-text: #6e7681; --color-checks-line-hover-bg: rgba(110,118,129,0.1); --color-checks-line-selected-bg: rgba(56,139,253,0.1); --color-checks-line-selected-num-text: #2f81f7; --color-checks-line-dt-fm-text: #ffffff; --color-checks-line-dt-fm-bg: #9e6a03; --color-checks-gate-bg: rgba(187,128,9,0.15); --color-checks-gate-text: #7d8590; --color-checks-gate-waiting-text: #d29922; --color-checks-step-header-open-bg: #161b22; --color-checks-step-error-text: #f85149; --color-checks-step-warning-text: #d29922; --color-checks-logline-text: #7d8590; --color-checks-logline-num-text: #6e7681; --color-checks-logline-debug-text: #a371f7; --color-checks-logline-error-text: #7d8590; --color-checks-logline-error-num-text: #6e7681; --color-checks-logline-error-bg: rgba(248,81,73,0.1); --color-checks-logline-warning-text: #7d8590; --color-checks-logline-warning-num-text: #d29922; --color-checks-logline-warning-bg: rgba(187,128,9,0.15); --color-checks-logline-command-text: #2f81f7; --color-checks-logline-section-text: #3fb950; --color-checks-ansi-black: #0d1117; --color-checks-ansi-black-bright: #161b22; --color-checks-ansi-white: #b1bac4; --color-checks-ansi-white-bright: #b1bac4; --color-checks-ansi-gray: #6e7681; --color-checks-ansi-red: #ff7b72; --color-checks-ansi-red-bright: #ffa198; --color-checks-ansi-green: #3fb950; --color-checks-ansi-green-bright: #56d364; --color-checks-ansi-yellow: #d29922; --color-checks-ansi-yellow-bright: #e3b341; --color-checks-ansi-blue: #58a6ff; --color-checks-ansi-blue-bright: #79c0ff; --color-checks-ansi-magenta: #bc8cff; --color-checks-ansi-magenta-bright: #d2a8ff; --color-checks-ansi-cyan: #76e3ea; --color-checks-ansi-cyan-bright: #b3f0ff; --color-mktg-btn-bg: #f6f8fa; --color-mktg-btn-shadow-focus: rgb(255 255 255 / 25%) 0 0 0 4px; --color-mktg-btn-shadow-hover: 0 4px 7px rgba(0, 0, 0, 0.15), 0 100px 80px rgba(255, 255, 255, 0.02), 0 42px 33px rgba(255, 255, 255, 0.024), 0 22px 18px rgba(255, 255, 255, 0.028), 0 12px 10px rgba(255, 255, 255, 0.034), 0 7px 5px rgba(255, 255, 255, 0.04), 0 3px 2px rgba(255, 255, 255, 0.07); --color-mktg-btn-shadow-hover-muted: rgb(255 255 255) 0 0 0 2px inset; --color-control-border-color-emphasis: #606771; --color-avatar-bg: rgba(255,255,255,0.1); --color-avatar-border: rgba(240,246,252,0.1); --color-avatar-stack-fade: #30363d; --color-avatar-stack-fade-more: #21262d; --color-avatar-child-shadow: 0 0 0 2px #0d1117; --color-topic-tag-border: rgba(0,0,0,0); --color-counter-border: rgba(0,0,0,0); --color-select-menu-backdrop-border: #484f58; --color-select-menu-tap-highlight: rgba(48,54,61,0.5); --color-select-menu-tap-focus-bg: #0c2d6b; --color-overlay-shadow: 0 0 0 1px #30363d, 0 16px 32px rgba(1,4,9,0.85); --color-overlay-backdrop: rgba(22,27,34,0.4); --color-header-text: rgba(255,255,255,0.7); --color-header-bg: #161b22; --color-header-divider: #8b949e; --color-header-logo: #f0f6fc; --color-header-search-bg: #0d1117; --color-header-search-border: #30363d; --color-sidenav-selected-bg: #21262d; --color-menu-bg-active: #161b22; --color-input-disabled-bg: rgba(110,118,129,0); --color-timeline-badge-bg: #21262d; --color-ansi-black: #484f58; --color-ansi-black-bright: #6e7681; --color-ansi-white: #b1bac4; --color-ansi-white-bright: #ffffff; --color-ansi-gray: #6e7681; --color-ansi-red: #ff7b72; --color-ansi-red-bright: #ffa198; --color-ansi-green: #3fb950; --color-ansi-green-bright: #56d364; --color-ansi-yellow: #d29922; --color-ansi-yellow-bright: #e3b341; --color-ansi-blue: #58a6ff; --color-ansi-blue-bright: #79c0ff; --color-ansi-magenta: #bc8cff; --color-ansi-magenta-bright: #d2a8ff; --color-ansi-cyan: #39c5cf; --color-ansi-cyan-bright: #56d4dd; --color-btn-text: #c9d1d9; --color-btn-bg: #21262d; --color-btn-border: rgba(240,246,252,0.1); --color-btn-shadow: 0 0 transparent; --color-btn-inset-shadow: 0 0 transparent; --color-btn-hover-bg: #30363d; --color-btn-hover-border: #8b949e; --color-btn-active-bg: hsla(212,12%,18%,1); --color-btn-active-border: #6e7681; --color-btn-selected-bg: #161b22; --color-btn-counter-bg: #30363d; --color-btn-primary-text: #ffffff; --color-btn-primary-bg: #238636; --color-btn-primary-border: rgba(240,246,252,0.1); --color-btn-primary-shadow: 0 0 transparent; --color-btn-primary-inset-shadow: 0 0 transparent; --color-btn-primary-hover-bg: #2ea043; --color-btn-primary-hover-border: rgba(240,246,252,0.1); --color-btn-primary-selected-bg: #238636; --color-btn-primary-selected-shadow: 0 0 transparent; --color-btn-primary-disabled-text: rgba(255,255,255,0.5); --color-btn-primary-disabled-bg: rgba(35,134,54,0.6); --color-btn-primary-disabled-border: rgba(240,246,252,0.1); --color-btn-primary-icon: #ffffff; --color-btn-primary-counter-bg: rgba(4,38,15,0.2); --color-btn-outline-text: #388bfd; --color-btn-outline-hover-text: #58a6ff; --color-btn-outline-hover-bg: #30363d; --color-btn-outline-hover-border: rgba(240,246,252,0.1); --color-btn-outline-hover-shadow: 0 1px 0 rgba(1,4,9,0.1); --color-btn-outline-hover-inset-shadow: inset 0 1px 0 rgba(255,255,255,0.03); --color-btn-outline-hover-counter-bg: rgba(5,29,77,0.2); --color-btn-outline-selected-text: #ffffff; --color-btn-outline-selected-bg: #0d419d; --color-btn-outline-selected-border: rgba(240,246,252,0.1); --color-btn-outline-selected-shadow: 0 0 transparent; --color-btn-outline-disabled-text: rgba(88,166,255,0.5); --color-btn-outline-disabled-bg: #0d1117; --color-btn-outline-disabled-counter-bg: rgba(31,111,235,0.05); --color-btn-outline-counter-bg: rgba(5,29,77,0.2); --color-btn-outline-hover-counter-fg: #58a6ff; --color-btn-outline-disabled-counter-fg: rgba(47,129,247,0.5); --color-btn-outline-counter-fg: #388bfd; --color-btn-danger-text: #f85149; --color-btn-danger-hover-text: #ffffff; --color-btn-danger-hover-bg: #da3633; --color-btn-danger-hover-border: #f85149; --color-btn-danger-hover-shadow: 0 0 transparent; --color-btn-danger-hover-inset-shadow: 0 0 transparent; --color-btn-danger-hover-icon: #ffffff; --color-btn-danger-hover-counter-bg: rgba(255,255,255,0.2); --color-btn-danger-selected-text: #ffffff; --color-btn-danger-selected-bg: #b62324; --color-btn-danger-selected-border: #ff7b72; --color-btn-danger-selected-shadow: 0 0 transparent; --color-btn-danger-disabled-text: rgba(248,81,73,0.5); --color-btn-danger-disabled-bg: #0d1117; --color-btn-danger-disabled-counter-bg: rgba(218,54,51,0.05); --color-btn-danger-counter-bg: rgba(73,2,2,0.2); --color-btn-danger-icon: #f85149; --color-btn-danger-counter-fg: #f85149; --color-btn-danger-disabled-counter-fg: rgba(248,81,73,0.5); --color-btn-danger-hover-counter-fg: #ffffff; --color-underlinenav-icon: #6e7681; --color-underlinenav-border-hover: rgba(110,118,129,0.4); --color-action-list-item-inline-divider: rgba(48,54,61,0.48); --color-action-list-item-default-hover-bg: rgba(177,186,196,0.12); --color-action-list-item-default-hover-border: rgba(0,0,0,0); --color-action-list-item-default-active-bg: rgba(177,186,196,0.2); --color-action-list-item-default-active-border: rgba(0,0,0,0); --color-action-list-item-default-selected-bg: rgba(177,186,196,0.08); --color-action-list-item-danger-hover-bg: rgba(248,81,73,0.16); --color-action-list-item-danger-active-bg: rgba(248,81,73,0.24); --color-action-list-item-danger-hover-text: #ff7b72; --color-switch-track-bg: rgba(110,118,129,0.1); --color-switch-track-hover-bg: hsla(215,8%,72%,0.1); --color-switch-track-active-bg: rgba(110,118,129,0.4); --color-switch-track-disabled-bg: #21262d; --color-switch-track-fg: #7d8590; --color-switch-track-disabled-fg: #010409; --color-switch-track-border: rgba(0,0,0,0); --color-switch-track-checked-bg: rgba(31,111,235,0.35); --color-switch-track-checked-hover-bg: rgba(31,111,235,0.5); --color-switch-track-checked-active-bg: rgba(31,111,235,0.65); --color-switch-track-checked-fg: #ffffff; --color-switch-track-checked-disabled-fg: #010409; --color-switch-track-checked-border: rgba(0,0,0,0); --color-switch-knob-bg: #0d1117; --color-switch-knob-border: #606771; --color-switch-knob-disabled-bg: #161b22; --color-switch-knob-checked-bg: #0d1117; --color-switch-knob-checked-disabled-bg: #161b22; --color-switch-knob-checked-border: rgba(31,111,235,0.35); --color-segmented-control-bg: rgba(110,118,129,0.1); --color-segmented-control-button-bg: #0d1117; --color-segmented-control-button-hover-bg: #30363d; --color-segmented-control-button-active-bg: #21262d; --color-segmented-control-button-selected-border: #6e7681; --color-tree-view-item-chevron-hover-bg: rgba(177,186,196,0.12); --color-tree-view-item-directory-fill: #7d8590; --color-fg-default: #e6edf3; --color-fg-muted: #7d8590; --color-fg-subtle: #6e7681; --color-fg-on-emphasis: #ffffff; --color-canvas-default: #0d1117; --color-canvas-overlay: #161b22; --color-canvas-inset: #010409; --color-canvas-subtle: #161b22; --color-border-default: #30363d; --color-border-muted: #21262d; --color-border-subtle: rgba(240,246,252,0.1); --color-shadow-small: 0 0 transparent; --color-shadow-medium: 0 3px 6px #010409; --color-shadow-large: 0 8px 24px #010409; --color-shadow-extra-large: 0 12px 48px #010409; --color-neutral-emphasis-plus: #6e7681; --color-neutral-emphasis: #6e7681; --color-neutral-muted: rgba(110,118,129,0.4); --color-neutral-subtle: rgba(110,118,129,0.1); --color-accent-fg: #2f81f7; --color-accent-emphasis: #1f6feb; --color-accent-muted: rgba(56,139,253,0.4); --color-accent-subtle: rgba(56,139,253,0.1); --color-success-fg: #3fb950; --color-success-emphasis: #238636; --color-success-muted: rgba(46,160,67,0.4); --color-success-subtle: rgba(46,160,67,0.15); --color-attention-fg: #d29922; --color-attention-emphasis: #9e6a03; --color-attention-muted: rgba(187,128,9,0.4); --color-attention-subtle: rgba(187,128,9,0.15); --color-severe-fg: #db6d28; --color-severe-emphasis: #bd561d; --color-severe-muted: rgba(219,109,40,0.4); --color-severe-subtle: rgba(219,109,40,0.1); --color-danger-fg: #f85149; --color-danger-emphasis: #da3633; --color-danger-muted: rgba(248,81,73,0.4); --color-danger-subtle: rgba(248,81,73,0.1); --color-open-fg: #3fb950; --color-open-emphasis: #238636; --color-open-muted: rgba(46,160,67,0.4); --color-open-subtle: rgba(46,160,67,0.15); --color-closed-fg: #f85149; --color-closed-emphasis: #da3633; --color-closed-muted: rgba(248,81,73,0.4); --color-closed-subtle: rgba(248,81,73,0.15); --color-done-fg: #a371f7; --color-done-emphasis: #8957e5; --color-done-muted: rgba(163,113,247,0.4); --color-done-subtle: rgba(163,113,247,0.1); --color-sponsors-fg: #db61a2; --color-sponsors-emphasis: #bf4b8a; --color-sponsors-muted: rgba(219,97,162,0.4); --color-sponsors-subtle: rgba(219,97,162,0.1); --color-primer-fg-disabled: #484f58; --color-primer-canvas-backdrop: rgba(1,4,9,0.8); --color-primer-canvas-sticky: rgba(13,17,23,0.95); --color-primer-border-active: #f78166; --color-primer-border-contrast: rgba(255,255,255,0.2); --color-primer-shadow-highlight: 0 0 transparent; --color-primer-shadow-inset: 0 0 transparent; --color-scale-black: #010409; --color-scale-white: #ffffff; --color-scale-gray-0: #f0f6fc; --color-scale-gray-1: #c9d1d9; --color-scale-gray-2: #b1bac4; --color-scale-gray-3: #8b949e; --color-scale-gray-4: #6e7681; --color-scale-gray-5: #484f58; --color-scale-gray-6: #30363d; --color-scale-gray-7: #21262d; --color-scale-gray-8: #161b22; --color-scale-gray-9: #0d1117; --color-scale-blue-0: #cae8ff; --color-scale-blue-1: #a5d6ff; --color-scale-blue-2: #79c0ff; --color-scale-blue-3: #58a6ff; --color-scale-blue-4: #388bfd; --color-scale-blue-5: #1f6feb; --color-scale-blue-6: #1158c7; --color-scale-blue-7: #0d419d; --color-scale-blue-8: #0c2d6b; --color-scale-blue-9: #051d4d; --color-scale-green-0: #aff5b4; --color-scale-green-1: #7ee787; --color-scale-green-2: #56d364; --color-scale-green-3: #3fb950; --color-scale-green-4: #2ea043; --color-scale-green-5: #238636; --color-scale-green-6: #196c2e; --color-scale-green-7: #0f5323; --color-scale-green-8: #033a16; --color-scale-green-9: #04260f; --color-scale-yellow-0: #f8e3a1; --color-scale-yellow-1: #f2cc60; --color-scale-yellow-2: #e3b341; --color-scale-yellow-3: #d29922; --color-scale-yellow-4: #bb8009; --color-scale-yellow-5: #9e6a03; --color-scale-yellow-6: #845306; --color-scale-yellow-7: #693e00; --color-scale-yellow-8: #4b2900; --color-scale-yellow-9: #341a00; --color-scale-orange-0: #ffdfb6; --color-scale-orange-1: #ffc680; --color-scale-orange-2: #ffa657; --color-scale-orange-3: #f0883e; --color-scale-orange-4: #db6d28; --color-scale-orange-5: #bd561d; --color-scale-orange-6: #9b4215; --color-scale-orange-7: #762d0a; --color-scale-orange-8: #5a1e02; --color-scale-orange-9: #3d1300; --color-scale-red-0: #ffdcd7; --color-scale-red-1: #ffc1ba; --color-scale-red-2: #ffa198; --color-scale-red-3: #ff7b72; --color-scale-red-4: #f85149; --color-scale-red-5: #da3633; --color-scale-red-6: #b62324; --color-scale-red-7: #8e1519; --color-scale-red-8: #67060c; --color-scale-red-9: #490202; --color-scale-purple-0: #eddeff; --color-scale-purple-1: #e2c5ff; --color-scale-purple-2: #d2a8ff; --color-scale-purple-3: #bc8cff; --color-scale-purple-4: #a371f7; --color-scale-purple-5: #8957e5; --color-scale-purple-6: #6e40c9; --color-scale-purple-7: #553098; --color-scale-purple-8: #3c1e70; --color-scale-purple-9: #271052; --color-scale-pink-0: #ffdaec; --color-scale-pink-1: #ffbedd; --color-scale-pink-2: #ff9bce; --color-scale-pink-3: #f778ba; --color-scale-pink-4: #db61a2; --color-scale-pink-5: #bf4b8a; --color-scale-pink-6: #9e3670; --color-scale-pink-7: #7d2457; --color-scale-pink-8: #5e103e; --color-scale-pink-9: #42062a; --color-scale-coral-0: #ffddd2; --color-scale-coral-1: #ffc2b2; --color-scale-coral-2: #ffa28b; --color-scale-coral-3: #f78166; --color-scale-coral-4: #ea6045; --color-scale-coral-5: #cf462d; --color-scale-coral-6: #ac3220; --color-scale-coral-7: #872012; --color-scale-coral-8: #640d04; --color-scale-coral-9: #460701;"><body class="logged-in env-production page-responsive intent-mouse" style="box-sizing: border-box; margin: 0px; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: var(--body-font-size, 14px); line-height: 1.5; color: var(--fgColor-default, var(--color-fg-default)); background-color: var(--bgColor-default, var(--color-canvas-default)); overflow-wrap: break-word;"><div data-turbo-body="" class="logged-in env-production page-responsive" style="box-sizing: border-box; isolation: isolate; overflow-wrap: break-word;"><div class="application-main " data-commit-hovercards-enabled="" data-discussion-hovercards-enabled="" data-issue-and-pr-hovercards-enabled="" style="box-sizing: border-box;"><div itemscope="" itemtype="http://schema.org/SoftwareSourceCode" class="" style="box-sizing: border-box;"><main id="js-repo-pjax-container" style="box-sizing: border-box; display: block;"><turbo-frame id="repo-content-turbo-frame" target="_top" data-turbo-action="advance" class="" style="box-sizing: border-box; display: block;"><div id="repo-content-pjax-container" class="repository-content " style="box-sizing: border-box;"><react-app app-name="react-code-view" initial-path="/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md" data-ssr="true" data-lazy="false" data-alternate="false" data-catalyst="" class="loaded" style="box-sizing: border-box; display: block; min-height: calc(100vh - 62px);"><div data-target="react-app.reactRoot" style="box-sizing: border-box;"><div class="Box-sc-g0xbh4-0" style="box-sizing: border-box;"><div class="Box-sc-g0xbh4-0 fSWWem" style="box-sizing: border-box; padding: 0px; --sticky-pane-height: 100vh;"><div class="Box-sc-g0xbh4-0 kPPmzM" style="box-sizing: border-box; max-width: 100%; margin-left: auto; margin-right: auto; display: flex; flex-wrap: wrap;"><div class="Box-sc-g0xbh4-0 cIAPDV" style="box-sizing: border-box; display: flex; flex: 1 1 100%; flex-wrap: wrap; max-width: 100%;"><main class="Box-sc-g0xbh4-0 emFMJu" style="box-sizing: border-box; display: flex; flex-direction: column; order: 2; flex: 1 1 0px; -webkit-box-flex: 1; min-width: 1px; margin-right: auto;"><div class="Box-sc-g0xbh4-0 hlUAHL" style="box-sizing: border-box; width: 1582px; max-width: 100%; margin-left: auto; margin-right: auto; -webkit-box-flex: 1; flex-grow: 1; padding: 0px;"><div data-selector="repos-split-pane-content" tabindex="0" class="Box-sc-g0xbh4-0 iStsmI" style="box-sizing: border-box; margin-left: auto; margin-right: auto; flex-direction: column; padding-bottom: 40px; max-width: 100%; margin-top: 0px; outline: none;"><div class="Box-sc-g0xbh4-0 MERGN" style="box-sizing: border-box; margin-left: 16px; margin-right: 16px;"><div class="Box-sc-g0xbh4-0 iJmJly" style="box-sizing: border-box; display: flex; flex-direction: row;"><div class="Box-sc-g0xbh4-0 jACbi container" style="box-sizing: border-box; width: 1550px; margin-right: 0px; margin-left: auto; container-type: inline-size; height: fit-content; min-width: 0px;"><div class="Box-sc-g0xbh4-0 etfROT" style="box-sizing: border-box; border-right: 1px solid rgb(48, 54, 61); border-bottom: 1px solid rgb(48, 54, 61); border-left: 1px solid rgb(48, 54, 61); border-image: initial; border-top: none rgb(48, 54, 61); border-radius: 0px 0px 6px 6px; min-width: 273px;"><section aria-labelledby="file-name-id-wide file-name-id-mobile" class="Box-sc-g0xbh4-0 iJJefD" style="box-sizing: border-box; display: block; background-color: var(--color-canvas-default); border: 0px; border-radius: 0px 0px 6px 6px; padding: 0px; min-width: 0px; margin-top: 46px; -webkit-box-pack: center; justify-content: center;"><div class="Box-sc-g0xbh4-0 cTsUqU js-snippet-clipboard-copy-unpositioned" data-hpc="true" style="box-sizing: border-box; border-bottom-left-radius: 6px; border-bottom-right-radius: 6px; padding: 32px; min-width: 0px;"><article class="markdown-body entry-content container-lg" itemprop="text" style="box-sizing: border-box; display: block; max-width: 1012px; margin-right: auto; margin-left: auto; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; line-height: 1.5; overflow-wrap: break-word;">

Name | Type | Description
-- | -- | --
Feature Type | uint8 | Set to value 5 to denote a Block Issuer Feature.
Expiry Slot | uint64 | The slot index at which the Block Issuer Feature expires and can be removed.
Block Issuer Keys Count | uint8 | The number of Block Issuer Keys.
Block Issuer Keys anyOf | Ed25519 Public Key Block Issuer KeyA Block Issuer Key derived from an Ed25519 Public Key. Defined in TIP-42 (Ed25519 Public Key Block Issuer Key).NameTypeDescriptionBlock Issuer Key Typeuint8Set to value 0 to denote an Ed25519 Public Key Block Issuer Key.Public KeyByteArray[32]The raw bytes of the Ed25519 public key. | Name | Type | Description | Block Issuer Key Type | uint8 | Set to value 0 to denote an Ed25519 Public Key Block Issuer Key. | Public Key | ByteArray[32] | The raw bytes of the Ed25519 public key.
Name | Type | Description
Block Issuer Key Type | uint8 | Set to value 0 to denote an Ed25519 Public Key Block Issuer Key.
Public Key | ByteArray[32] | The raw bytes of the Ed25519 public key.

<h1 tabindex="-1" id="user-content-migration" dir="auto" style="box-sizing: border-box; font-size: 2em; margin: 24px 0px 16px; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a class="heading-link" href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#migration" style="box-sizing: border-box; background-color: transparent; color: unset; text-decoration: none;">Migration<svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg></a></h1><h2 tabindex="-1" id="user-content-alias-to-account-conversion" dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.5em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a class="heading-link" href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#alias-to-account-conversion" style="box-sizing: border-box; background-color: transparent; color: unset; text-decoration: none;">Alias to Account Conversion<svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg></a></h2><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><em style="box-sizing: border-box;">Alias Outputs</em><span> </span>in Stardust-based networks are converted to<span> </span><em style="box-sizing: border-box;">Account Outputs</em>. Account Outputs' BIC balance is set to<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">0</code>.</p><h2 tabindex="-1" id="user-content-block-issuer-accounts" dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.5em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a class="heading-link" href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#block-issuer-accounts" style="box-sizing: border-box; background-color: transparent; color: unset; text-decoration: none;">Block Issuer Accounts<svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg></a></h2><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Since IOTA 2.0 requires Accounts with a Block Issuer Feature to issue blocks and Stardust-based networks do not have these, a migration procedure is required so the network can be bootstrapped. Users can add a serialized Block Issuer Feature to an<span> </span><em style="box-sizing: border-box;">Alias Output</em><span> </span>which is converted to an<span> </span><em style="box-sizing: border-box;">Account Output</em><span> </span>with a Block Issuer Feature and the Metadata Feature is removed from the output. This transition is only done if all of the following conditions hold:</p><ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><li style="box-sizing: border-box;">The<span> </span><em style="box-sizing: border-box;">Alias Output</em>'s<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">Amount</code><span> </span>is at least<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">Rent Parameters::Storage Cost * Account Output Score</code>, with the latter defined in<span> </span><a href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#storage-deposit" style="box-sizing: border-box; background-color: transparent; color: var(--fgColor-accent, var(--color-accent-fg)); text-decoration: none;">Implicit Account (Storage Deposit)</a>, except that the<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">Block Issuer Keys Count</code><span> </span>of the output for the deposit calculation must be set to the value of the<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">Block Issuer Keys Count</code><span> </span>read from the serialized Block Issuer Feature in the to-be-converted<span> </span><em style="box-sizing: border-box;">Alias Output</em>.</li><li style="box-sizing: border-box; margin-top: 0.25em;">The serialized Block Issuer Feature is prefixed with the ascii-encoded string<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">BlockIssuerFeature</code>, which is<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">[0x42, 0x6c, 0x6f, 0x63, 0x6b, 0x49, 0x73, 0x73, 0x75, 0x65, 0x72, 0x46, 0x65, 0x61, 0x74, 0x75, 0x72, 0x65]</code><span> </span>in hex encoding, expressed as a list in JSON.</li><li style="box-sizing: border-box; margin-top: 0.25em;">The Block Issuer Feature must pass syntactic validation.</li></ul><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">If not all conditions hold, the converted Account Output will contain the Metadata Feature untouched.</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Note: It is recommended to set the<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">Expiry Slot</code><span> </span>to the maximum value of a<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">uint32</code>.</p><h1 tabindex="-1" id="user-content-rationale--alternatives" dir="auto" style="box-sizing: border-box; font-size: 2em; margin: 24px 0px 16px; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a class="heading-link" href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#rationale--alternatives" style="box-sizing: border-box; background-color: transparent; color: unset; text-decoration: none;">Rationale &amp; Alternatives<svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg></a></h1><h2 tabindex="-1" id="user-content-implicit-accounts" dir="auto" style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.5em; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a class="heading-link" href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#implicit-accounts" style="box-sizing: border-box; background-color: transparent; color: unset; text-decoration: none;">Implicit Accounts<svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg></a></h2><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The rationale for having implicit accounts is that first-time users in a network usually receive funds in Basic Outputs rather than Account Outputs. This leaves them unable to issue blocks since an Account with a Block Issuer Feature is required to do so. Implicit Accounts give those users a permissionless way to create their account with the only requirement being owned funds in an output.</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Implicit Account Creation Address is a separate type of address rather than a<span> </span><a href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0050/tip-0050.md#capability-flags" style="box-sizing: border-box; background-color: transparent; color: var(--fgColor-accent, var(--color-accent-fg)); text-decoration: none;">capability</a><span> </span>because it implies many of the capability flags but also has additional non-trivial transaction validation rules that don't fit together with the other boolean flags that are only concerned about one aspect of transaction validity. It would also have implications for address validity. For instance, it would be invalid for an NFT Address to have the flag for implicit account creation set. Other capability flags do not have implications for address validity. Moreover, the implicit account creation address can only be used in a very limited context, i.e. within an Address Unlock Condition in a Basic Output. These reasons warrant a separate address type.</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">TODO: Describe utility of max<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">uint32</code><span> </span>values for<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">Expiry Slot</code><span> </span>and<span> </span><code style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 13.6px; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: var(--bgColor-neutral-muted, var(--color-neutral-muted)); border-radius: 6px;">End Epoch</code>.</p><h1 tabindex="-1" id="user-content-copyright" dir="auto" style="box-sizing: border-box; font-size: 2em; margin: 24px 0px 16px; font-weight: var(--base-text-weight-semibold, 600); line-height: 1.25; padding-bottom: 0.3em; border-bottom: 1px solid var(--borderColor-muted, var(--color-border-muted)); color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><a class="heading-link" href="https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#copyright" style="box-sizing: border-box; background-color: transparent; color: unset; text-decoration: none;">Copyright<svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path d="m7.775 3.275 1.25-1.25a3.5 3.5 0 1 1 4.95 4.95l-2.5 2.5a3.5 3.5 0 0 1-4.95 0 .751.751 0 0 1 .018-1.042.751.751 0 0 1 1.042-.018 1.998 1.998 0 0 0 2.83 0l2.5-2.5a2.002 2.002 0 0 0-2.83-2.83l-1.25 1.25a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042Zm-4.69 9.64a1.998 1.998 0 0 0 2.83 0l1.25-1.25a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042l-1.25 1.25a3.5 3.5 0 1 1-4.95-4.95l2.5-2.5a3.5 3.5 0 0 1 4.95 0 .751.751 0 0 1-.018 1.042.751.751 0 0 1-1.042.018 1.998 1.998 0 0 0-2.83 0l-2.5 2.5a1.998 1.998 0 0 0 0 2.83Z"></path></svg></a></h1><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px !important; color: rgb(230, 237, 243); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(13, 17, 23); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Copyright and related rights waived via<span> </span><a href="https://creativecommons.org/publicdomain/zero/1.0/" rel="nofollow" style="box-sizing: border-box; background-color: transparent; color: var(--fgColor-accent, var(--color-accent-fg)); text-decoration: none;">CC0</a>.</p></article></div></section></div></div></div></div></div></div><div class="Box-sc-g0xbh4-0" style="box-sizing: border-box;"></div></main></div></div></div><div id="find-result-marks-container" class="Box-sc-g0xbh4-0 aZrVR" style="box-sizing: border-box; position: fixed; top: 0px; right: 0px; height: 970px; width: 15px; transition: transform 0.3s ease 0s; z-index: 1;"></div></div></div></react-app></div></turbo-frame></main></div></div></div><div id="js-global-screen-reader-notice" class="sr-only" aria-live="polite" style="box-sizing: border-box; position: absolute; width: 1px; height: 1px; padding: 0px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); overflow-wrap: normal; border: 0px;">tips-draft/tips/TIP-0042/tip-0042.md at tip42 · iotaledger/tips-draft</div></body></html><!--EndFragment-->
</body>
</html>Name	Type	Description
Feature Type	uint8	Set to value 5 to denote a Block Issuer Feature.
Expiry Slot	uint64	The slot index at which the Block Issuer Feature expires and can be removed.
Block Issuer Keys Count	uint8	The number of Block Issuer Keys.
Block Issuer Keys anyOf	
Ed25519 Public Key Block Issuer Key
A Block Issuer Key derived from an Ed25519 Public Key. Defined in [TIP-42 (Ed25519 Public Key Block Issuer Key)](https://github.com/iotaledger/tips-draft/blob/tip42/tips/TIP-0042/tip-0042.md#ed25519-public-key-block-issuer-key).
Name	Type	Description
Block Issuer Key Type	uint8	Set to value 0 to denote an Ed25519 Public Key Block Issuer Key.
Public Key	ByteArray[32]	The raw bytes of the Ed25519 public key.

Reply:
https://github.com/iotaledger/tips-draft/pull/11#discussion_r1352622916

Reply:
Changed